options.c

```diff
@@ -504,6 +506,54 @@ static void set_tree_width_max(void *data, const char *buf)
 	update_size();
 }
 
+/* get_pl_env_vars converts the pl_env_vars array into a comma-separated list */
+static void get_pl_env_vars(void *data, char *buf, size_t size)
+{
+	if (!pl_env_vars) {
+		strscpy(buf, "", size);
+		return;
+	}
+	char *p = buf;
+	size_t r = size - 1;
+	for (char **x = pl_env_vars; *x; x++) {
+		if (x != pl_env_vars) {
+			if (!--r)
+				return;
+			*p++ = ',';
+		}
+		size_t l = strlen(*x);
+		if (!(r -= l))
```

Comment:
Hmm, am I wrong or is there no guarantee we will hit 0 here (or in the `--r` case above)?

Reply:
That's correct. `r` is only there to prevent the buffer from overflowing (most of the other setting functions also just cut it off). The for loop condition is what will prevent us from going past the end of the string, since it'll end when the string hits the `NULL` terminator.

Reply:
Comment added.